torch/fx/passes/_tensorify_python_scalars.py

```diff
@@ -281,6 +288,37 @@ def _sympy_interp(expr: sympy.Expr) -> MetaProxy:
                     node.replace_all_uses_with(guard_scalar(val))
                     graph.erase_node(node)
 
+    should_restart = False
+    for i, node in enumerate(graph.nodes):
+        if (
+            node.op == "placeholder"
+            and len(node.users) == 0
+            and (name := placeholder_to_symfloat_name.get(node))
+        ):
+            # At this point we've lost the back pointer to
+            # what f_local this placeholder points to. Instead,
+            # we will rely on the symfloat name to specialize when we
+            # restart analysis.
+            TensorifyState.specialize(name)
+            should_restart = True
+
+    # Sometimes by the time we get to tensorify, there have already been
```

Comment:
This is the meat of the PR. Because by the time we reach tensorify we've lost all source/name information in the fx graph, I utilize shape_env to figure out which specializations have already happened.